src/DynamoCoreWpf/ViewModels/Core/InPortViewModel.cs

```diff
@@ -317,16 +329,13 @@ protected override void RefreshPortColors()
 
         internal void RefreshInputPortsByOutputConnectionChanged(bool isOutputConnected)
         {
-            if (node.NodeModel.IsPartiallyApplied)
+            if (node.NodeModel.AreAllInputsDisconnected && isOutputConnected)
```

Comment:
Do we still need the old check with these two?

Reply:
Not in this case because is related to the output, where the inputs need to be gray if all are disconnected and the output is connected.